### PR TITLE
docs: Add S3 compliant storage

### PR DESCRIPTION
**Pull request by Dan-Heath**
This PR updates the session recording docs to include implied support for S3-compliant storage providers. Refer to the following topics from the preview deployment:

- [Overview](https://boundary-g2tbzm4ji-hashicorp.vercel.app/boundary/docs/configuration/session-recording)
- [Configure an S3-compliant storage provider](https://boundary-g2tbzm4ji-hashicorp.vercel.app/boundary/docs/configuration/session-recording/storage-providers/configure-s3-compliant)
- [Create storage buckets](https://boundary-g2tbzm4ji-hashicorp.vercel.app/boundary/docs/configuration/session-recording/create-storage-bucket)

### CODE REVIEW

File: website/content/docs/configuration/session-recording/create-storage-bucket.mdx [hunk 1]
```diff
@@ -314,6 +315,84 @@ Complete the following steps to create a storage bucket in Boundary.
 </Tab>
 </Tabs>
 
+</Tab>
+
+<Tab heading="S3-compliant">
+
+Complete the following steps to create a storage bucket in Boundary.
```
Comment (stellarsquall) [inline website/content/docs/configuration/session-recording/create-storage-bucket.mdx:322]: not sure if this is the right place for this, but we probably want to call out that we're using hitachi as the example 

```suggestion
Complete the following steps to create a storage bucket in Boundary using an S3-compliant storage provider. Hitachi Content Platform is used as an example below.
```

File: website/content/docs/configuration/session-recording/storage-providers/configure-s3-compliant.mdx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+<EnterpriseAlert product="boundary">This feature requires <a href="https://www.hashicorp.com/products/boundary">HCP Boundary or Boundary Enterprise</a></EnterpriseAlert>
+
+The [MinIO plugin](https://github.com/hashicorp/boundary-plugin-minio/) lets you configure S3-compliant storage providers for session recording.
+
+HashiCorp has tested and confirmed that you can configure the following S3-compliant storage products for session recording using the MinIO plugin:
+
+- [Hitachi Content Platform](#hitachi-content-platform-configuration)
+
+It should also be possible to configure other S3-compliant storage providers to work for session recording storage, but we have not tested other providers.
```
Comment (anando-chatterjee) [inline website/content/docs/configuration/session-recording/storage-providers/configure-s3-compliant.mdx:18]: IMHO we should encourage users to try out other S3 compliant providers. Might want to change this up a little along the lines of:

> Other S3 compliant object storage providers can also be configured for session recording storage and we will add to the list of tested providers as we get around to testing other providers.

File: website/content/docs/configuration/session-recording/create-storage-bucket.mdx [hunk 3]
```diff
@@ -314,6 +315,84 @@ Complete the following steps to create a storage bucket in Boundary.
...
+   - `secret`: (Required) The storage provider's credentials to use.
+      - `access_key_id` (Required): The storage provider's service account's access key to use with this storage bucket.
+      - `secret_access_key` (Required): The storage provider's service account's secret key to use with this storage bucket.
+   - `attributes` or `-attr`: Attributes of the S3-compliant storage bucket.
+      - `endpoint_url` (Required): Fully-qualified endpoint pointing to an S3-compliant API. This example uses Hitachi, but you should substitute your storage provider's endpoint.
+      - `region`: (Optional) The region to configure the storage bucket for.
+      - `disable_credential_rotation`: (Optional) Controls whether the plugin will rotate the incoming credentials and manage a new storage service account. If this attribute is set to `false`, or not provided, the plugin will rotate the incoming credentials to create a new storage service account, then delete the incoming credentials.
+
+      This option must be set to `true` if you use dynamic credentials.
```
Comment (hugoghx) [inline website/content/docs/configuration/session-recording/create-storage-bucket.mdx:390]: We only support dynamic credentials for AWS - Other services don't really have this at all, so I think we could get rid of this line given the context in this page?

```suggestion

```

**Review by louisruch — APPROVED** [review]
This looks great

**Pull request merged**